chainerx_cc/chainerx/routines/misc.cc

```diff
@@ -116,6 +116,11 @@ void MaximumImpl(const Array& x1, const Array& x2, const Array& out) { IfGreater
 
 }  // namespace
 
+Array IfGreaterElse(const Array& x1, const Array& x2, const Array& pos, const Array& neg, const Array& out) {
+    IfGreaterElseImpl(x1, x2, pos, neg, out);
+    return out;
+}
```

Comment:
I guess the signature of the function should be
`void IfGreaterElse(const Array& x1, const Array& x2, const Array& pos, const Array& neg, const Array& out)`
with current implementation we mostly return a copy of `out` as well as alter the referenced argument as well. (Not sure)

Or alternatively don't take `out` as input, but initialise `out` in function and return it.
Current implementation is a bit confusing to me.
@hvy your opinion?

Reply:
For Elu, we're actually looking for a ASAA? Since zero can be a scalar. But AAAA is also fine for now if that allows us to reuse existing code. And, if this is to be exposed as a routine, yes out should be allocated inside the routine and returned.